### PR TITLE
chore: release fvm v3.10.0

### PR DESCRIPTION
**Pull request by rjan90**
Update `filecoin-proofs-api` to v18

### CODE REVIEW

**Comment by Stebalien** [comment]
Ah, great. It looks like we need to backport https://github.com/filecoin-project/ref-fvm/pull/2009 as well.

**Comment by rjan90** [comment]
> Ah, great. It looks like we need to backport https://github.com/filecoin-project/ref-fvm/pull/2009 as well.

Backported and updated in Changelog now.

File: testing/integration/Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ authors = ["Protocol Labs", "Filecoin Core Devs", "Polyphene"]
 repository = "https://github.com/filecoin-project/ref-fvm"
 
 [dependencies]
-fvm = { version = "3.9.0", path = "../../fvm", default-features = false, features = ["testing"] }
+fvm = { version = "3.10.0", path = "../../fvm", default-features = false, features = ["testing"] }
 fvm_shared = { version = "3.6.0", path = "../../shared", features = ["testing"] }
```
Comment (Stebalien) [inline testing/integration/Cargo.toml:12]: need to update fvm_shared. Unlike in v4, in v3 we're not (yet) using the workspace itself to track versions (patches welcome).

Comment (Stebalien) [reply testing/integration/Cargo.toml:12]: Also need to release shared.

Comment (rjan90) [reply testing/integration/Cargo.toml]: I updated fvm_shared to 3.1.0.0 in https://github.com/filecoin-project/ref-fvm/pull/2016/commits/1024200283a93f028e949cbe15fd4d750a5a1ca3, but it seems like that caused CI to fail. So I thought I had to run `cargo check --all` but that failed with:

```
error: failed to select a version for the requirement `fvm_shared = "^3.10.0"`
candidate versions found which didn't match: 3.6.0
location searched: /Users/phi/ref-fvm/shared
required by package `fvm_integration_tests v3.3.0 (/Users/phi/ref-fvm/testing/integration)`
```

Any ideas for how I can get this to pass after updating fvm_shared? Sorry for asking so many novice questions, not that familiar with ref-fvm.

Comment (Stebalien) [reply testing/integration/Cargo.toml]: You need to update the `fvm_shared` version. Unlike in master (v4), we haven't moved the versions to the root cargo.toml.

Comment (Stebalien) [reply testing/integration/Cargo.toml]: (so `fvm` and `fvm_shared` can actually have different versions)

Comment (Stebalien) [reply testing/integration/Cargo.toml]: (if you have some time and want to backport the workspace versioning changes from master, I'd appreciate it)

**Comment by Stebalien** [comment]
We also need an update to the `fvm_shared` changelog.

**Review by Stebalien — APPROVED** [review]

**Comment by rjan90** [comment]
> We also need an update to the fvm_shared changelog.

Added in https://github.com/filecoin-project/ref-fvm/commit/c6d33d8140ae7bdd359404d9c6f4d4705983c9da

**Pull request merged**